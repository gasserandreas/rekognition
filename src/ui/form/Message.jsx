import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Colors } from '../../styles';

export const MESSAGE_TYPES = {
  INFO: 'info',
  ERROR: 'error',
  WARNING: 'warning',
};

const getColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.ColorsPalette.TextInvers;
    default:
      return Colors.ColorsPalette.Text;
  }
}

const getBackgroundColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.Red.Default;
    case MESSAGE_TYPES.WARNING:
      return Colors.Orange.Default;
    default:
      return 'inherit';
  }
}

const getBorderColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.Red.Default;
    case MESSAGE_TYPES.WARNING:
      return Colors.Orange.Default;
    default:
      return Colors.Neutrals.LightDark;
  }
}

/* istanbul ignore next */
const StyledMessage = styled.div`
  color: ${props => getColor(props)};
  background-color: ${props => getBackgroundColor(props)};
  border-color: ${props => getBorderColor(props)};
  border-width: ${props => props.border ? '1px' : '0'};
  border-style: solid;
  border-radius: ${props => props.rounded ? '2px' : '0'};
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
`;

const Message = ({ children, ...props}) => (
  <StyledMessage {...props}>
    {children}
  </StyledMessage>
);

Message.propTypes = {
  appearance: PropTypes.oneOf([
    MESSAGE_TYPES.INFO,
    MESSAGE_TYPES.ERROR,
    MESSAGE_TYPES.WARNING,
  ]),
  children: PropTypes.node.isRequired,
  border: PropTypes.bool,
  rounded: PropTypes.bool,
};

Message.defaultProps = {
  appearance: MESSAGE_TYPES.INFO,
  border: false,
  rounded: true,
};

export const __testables__ = {
  getColor,
  getBorderColor,
  getBackgroundColor,
  StyledMessage,
};

export default Message;
