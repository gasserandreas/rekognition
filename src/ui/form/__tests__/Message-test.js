/* global it, React, shallow, mount, render, renderer, testSnapshot */

import Message, { MESSAGE_TYPES } from '../Message';

it('Message should render correctly', () => {
  testSnapshot(<Message><p>Hello</p></Message>);
});

it('Message should render info styles', () => {
  testSnapshot(<Message appearance={MESSAGE_TYPES.INFO}><p>Hello</p></Message>);
});

it('Message should render warning styles', () => {
  testSnapshot(<Message appearance={MESSAGE_TYPES.WARNING}><p>Hello</p></Message>);
});

it('Message should render error styles', () => {
  testSnapshot(<Message appearance={MESSAGE_TYPES.ERROR}><p>Hello</p></Message>);
});

it('Message should render without border', () => {
  testSnapshot(<Message border><p>Hello</p></Message>);
});

it('Message should render without round corners', () => {
  testSnapshot(<Message rounded={false}><p>Hello</p></Message>);
});

