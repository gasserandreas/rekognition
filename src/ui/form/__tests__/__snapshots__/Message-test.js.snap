// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
<div
  className="sc-bdVaJa eemERo"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Message should render error styles 1`] = `
<div
  className="sc-bdVaJa gQuyMB"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Message should render info styles 1`] = `
<div
  className="sc-bdVaJa eemERo"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Message should render warning styles 1`] = `
<div
  className="sc-bdVaJa fQxPiK"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Message should render without border 1`] = `
<div
  className="sc-bdVaJa jwkoFj"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Message should render without round corners 1`] = `
<div
  className="sc-bdVaJa jBXRes"
>
  <p>
    Hello
  </p>
</div>
`;
